### ISSUE TITLE
Adding a Solana Wallet OAuth

### ISSUE THREAD

**Issue by Arunsoniiii** [code100x/github-sol-auto-dispenser#28]
We can have a wallet OAuth option for solana based wallets.

![image](https://github.com/code100x/github-sol-auto-dispenser/assets/149182931/4edfcb3c-57eb-480e-a6ea-f40204ffe0a9)

Hey @devsargam @hkirat can i work on that?

**Comment by devsargam** [comment]
@Arunsoniiii yep sure go on!!!

**Comment by Arunsoniiii** [comment]
Thanks

**Comment by Arunsoniiii** [comment]
Hey @devsargam where do we exactly need select wallet button in oAuth or can we have a saperate button ?

**Comment by devsargam** [comment]
Can do that but also can u make a fallback if someone doesn't have backpack acc still giving a solona address can let u claim the bounty. Make it urself no forced decision. Nice design + functional works

**Comment by Arunsoniiii** [comment]
cool thingy, 
[Screencast from 15-04-24 12_50_05 AM IST.webm](https://github.com/code100x/github-sol-auto-dispenser/assets/149182931/dd3541c0-1495-406d-b827-32cbb32bd35d)

I made this, when we click on select wallet it'll automatically detect wallet and conncet to the wallet also if it handles all the problems, and suggestion: we can this in navbar. 

please let me know how's it? @devsargam

**Comment by devsargam** [comment]
LGTM can u pr it?

**Comment by Arunsoniiii** [comment]
wait sir video isn't good let me reshare 

https://github.com/code100x/github-sol-auto-dispenser/assets/149182931/eb1f7a37-d1b6-4f39-94b3-bf9575785eef

**Comment by devsargam** [comment]
Works but @Arunsoniiii keep in mind we don't want to confuse contributors (most of them don't know what the hell a wallet is). They should I ideally be able to put in their solona address and be gtg.

**Comment by Arunsoniiii** [comment]
yeah I know that's why we can have saperate step where a contributor connect their wallet should be in navbar aur should be there, where they are claiming bounty @devsargam

**Comment by devsargam** [comment]
The logic is in contributors-web app

Also, migration of that solola to separate package could be a good thing

**Comment by Arunsoniiii** [comment]
@devsargam I am facing access related error since past 2 days how can i fix this?

**Comment by devsargam** [comment]
@Arunsoniiii access as in? Can u try making ur github mail public and then try once?

**Comment by Arunsoniiii** [comment]
> @Arunsoniiii access as in? Can u try making ur github mail public and then try once?

Actully i'm getting error related to repo access so i thought I need ssh
